app/Services/ScrapeService.php

```diff
@@ -0,0 +1,227 @@
+<?php
+namespace App\Services;
+
+use Exception;
+use Facebook\WebDriver\Exception\NoSuchElementException;
+use Facebook\WebDriver\Exception\TimeoutException;
+use Facebook\WebDriver\WebDriverDimension;
+use Facebook\WebDriver\WebDriverElement;
+use Symfony\Component\Panther\Client;
+use Symfony\Component\DomCrawler\Crawler;
+use App\Helpers\Building;
+
+class ScrapeService {
+    private Client $client;
+
+    function __construct()
+    {
+        $options = config('scraper.client');
+
+        $this->client = Client::createSeleniumClient($options['host']);
+
+        $webDriver = $this->client->getWebDriver();
+        $webDriver->manage()->window()->setSize(new WebDriverDimension($options['width'], $options['height']));
+    }
+
+    /**
+     * @throws NoSuchElementException
+     * @throws TimeoutException
+     */
+    public function initialFetch(string $country, int $quantity): void
+    {
+        $count = 0;
+        $page = 1;
+
+        $this->client->request('GET', config('scraper.client.trivago_url'));
+
+        $directoryName = config('scraper.command.scraper_directory') . strtotime('now') . '_' . $country;
+        mkdir($directoryName, recursive: true);
+
+        while ($count < $quantity) {
+            if ($page == 1 && $count == 0) {
+                $this->selectCountry($country);
+
+                $submit = $this->client->waitFor('[data-testid="search-button-with-loader"]');
+                // First click to confirm choice, Second to redirect
+                $submit->filter('[data-testid="search-button-with-loader"]')
+                    ->click();
+                $submit->filter('[data-testid="search-button-with-loader"]')
+                    ->click();
+            } else {
+                $url = $this->client->getCurrentURL() . ';pa-' . $page;
+                $this->client->request('GET', $url);
+            }
+
+            $this->client->waitFor('[data-testid="accommodation-list"]', config('scraper.crawler.timeouts.long'));
+
+            for ($i = 1; $i < config('scraper.crawler.hotels_per_page'); $i++) {
+                if ($count == $quantity)
+                    die();
+
+                try {
+                    $building = $this->fetchBuilding($i);
+                    $this->appendToFiles($building, $directoryName);
+
+                    $count++;
+                } catch (Exception) {
+                    echo "Ommitted a building: Failed to scrape building $i on page $page.\n";
+                }
+            }
+            $page++;
+        }
+    }
```

Comment:
I mean... This is fine, but I just wanted to say that if you don't want to add a try-catch to `ScrapeCommand.php`, you can remove the PHPDoc before the function and just surround `->waitFor()` calls with try-catches, which seems kind of better to me.